### PR TITLE
set-up-endpoints-volunter-and-post-endpoints

### LINKED ISSUE #49: set up endpoints: getVolunteersByEvent, createVolunteerAtEvent, createPostEvent, getPostEvent
Set up the following endpoints:
- getVolunteersByEvent
- createVolunteerAtEvent
- createPostEvent
- getPostEvent

### PR DESCRIPTION
**Pull request by ctc-devops**
Set up the following endpoints:
- getVolunteersByEvent
- createVolunteerAtEvent
- createPostEvent
- getPostEvent
closes #49

### CODE REVIEW

File: routes/volunteer_at_events/volunteer_at_events.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// endpoints related to volunteer_at_events
+const express = require('express');
+const pool = require('../../db');
+
+const volunteerAtEventsRouter = express();
+
+volunteerAtEventsRouter.use(express.json());
+
+// Get Volunteers By Event
+volunteerAtEventsRouter.get('/:id', async (req, res) => {
```
Comment (henryg0) [inline routes/volunteer_at_events/volunteer_at_events.js]: move to eventsRouter under route: '/events/:id/volunteers', don't forget to remove '/events' from the router.get() call

File: routes/volunteer_at_events/volunteer_at_events.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      [req.params.id],
+    );
+    res.status(200).json(getVolunteerByEvent.rows);
+  } catch (err) {
+    res.status(400).json(err);
+  }
+});
+
+// Create Volunteer At Event
+volunteerAtEventsRouter.post('/create', async (req, res) => {
```
Comment (henryg0) [inline routes/volunteer_at_events/volunteer_at_events.js]: move to volunteersRouter under route '/volunteers/event/join', don't forget to remove the '/volunteers' from the router.post() call

**Comment by Madhu2244** [comment]
We were not able to test the add volunteer_at_event router, but the other ones are working!

File: routes/postevents/postevents.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  } catch (err) {
+    res.status(400).json(err);
+  }
+});
+
+// Create Post Event
+posteventRouter.post('/create', async (req, res) => {
+  try {
+    const { eventId, description } = req.body;
+    const createPostEvent = await pool.query(
```
Comment (henryg0) [inline routes/postevents/postevents.js:31]: add query that checks if postevent already exists for event, if so then don't create one

File: routes/volunteers/volunteers.js [hunk 4]
```diff
@@ -45,4 +44,18 @@ volunteerRouter.get('/available', async (req, res) => {
   }
 });
 
+// Create Volunteer At Event
+volunteerRouter.post('/event/join', async (req, res) => {
+  try {
+    const { userId, eventId, notes } = req.body;
```
Comment (henryg0) [inline routes/volunteers/volunteers.js:50]: use req params

**Review by henryg0 — APPROVED** [review]

**Pull request merged**